### PR TITLE
commenting rate limit for users who have received a high % of downvotes

### PR DESCRIPTION
**Pull request by s-cheng**
This PR is a follow up to https://github.com/ForumMagnum/ForumMagnum/pull/7026. This adds a second condition that can trigger the commenting rate limit: if the user has received a high % of downvotes (EA Forum will set it at 30%).

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204541387983665) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by s-cheng** [comment]
@Raemon You can see some of the specific data that Clifford used to decide in [his doc here](https://docs.google.com/document/d/1Fmg5lL_91-5O-4Hp1sX1NN5FX630UIOvQurpvWC65cM/edit#) - I think the gist is that the rate limit is mild and it's rare for users >30 karma to get that many downvotes, though we'll check to see how many people this actually affects. Appreciate the feedback though, I'll flag it to him.

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 1]
```diff
@@ -2635,7 +2665,7 @@ const schema: SchemaType<DbUser> = {
   },
   
   rateLimitNextAbleToComment: {
-    type: Date,
+    type: GraphQLJSON,
```
Comment (s-cheng) [inline packages/lesswrong/lib/collections/users/schema.ts:2668]: There might be a better type to use here, let me know if you have suggestions

**Comment by imbenclifford** [comment]
Thanks @Raemon! I did consider something like this when (I think) you mentioned it before and I do think it might make sense. It's just as Sarah said that when I looked at the data at the time, this worked pretty well.

File: packages/lesswrong/lib/collections/users/schema.ts (hunk 1 above)
Comment (Will-Howard) [reply packages/lesswrong/lib/collections/users/schema.ts:2668]: We have a function `simplSchemaToGraphQLtype` but it looks like it doesn't actually do anything currently :(

File: packages/lesswrong/lib/sql/Query.ts [hunk 2]
```diff
@@ -394,14 +395,13 @@ abstract class Query<T extends DbObject> {
...
-           * We filter out the hints because they'll incorrectly be `::TEXT[]`, rather than `::TEXT`
-           * That's because we only take the subtype above for $all comparers.  (Not sure why, so...)
+           * We use the original type hint, rather then subtype, because otherwise array fields will have the wrong hint
            * 
            * We use `&&` to do an intersection ("any values in the field match any values passed in") rather than "contains the entire subset"
            * As far as I can tell this case is only used for meetup types and we should avoid doing this elsewhere (and just hand-write some SQL)
            */
           if (fieldType?.isArray()) {
-            return [field, hint, "&& ARRAY[", ...args.filter((arg) => arg !== hint), "]"]
+            return [field, originalFieldTypeHint, "&& ARRAY[", ...args, "]"]
```
Comment (Will-Howard) [inline packages/lesswrong/lib/sql/Query.ts:404]: Thank for fixing this

**Review by Will-Howard — APPROVED** [review]
🦭

**Pull request merged**